src/core_atmosphere/physics/physics_noaa/SMOKE/module_add_emiss_burn.F90

```diff
@@ -0,0 +1,181 @@
+                           swdown,ebb_dcycle, ebu,fire_type,&
+                           q_vap, add_fire_moist_flux,              &
+                           sc_factor,                               &
+                           ids,ide, jds,jde, kds,kde,               &
+                           ims,ime, jms,jme, kms,kme,               &
+                           its,ite, jts,jte, kts,kte                )
+
+   IMPLICIT NONE
+
+   INTEGER,      INTENT(IN   ) :: julday,                           &
+                                  ids,ide, jds,jde, kds,kde,        &
+                                  ims,ime, jms,jme, kms,kme,        &
+                                  its,ite, jts,jte, kts,kte
+
+   real(RKIND), DIMENSION( ims:ime, kms:kme, jms:jme ),                 &
+         INTENT(INOUT ) ::                                   smoke!, smoke_coarse 
+
+   real(RKIND), DIMENSION( ims:ime, kms:kme, jms:jme ),                 &
+         INTENT(INOUT ) ::                                   ebu, q_vap ! SRB: added q_vap
+         !INTENT(INOUT ) ::                                   ebu, ebu_coarse, q_vap ! SRB: added q_vap
+
+   real(RKIND), DIMENSION(ims:ime,jms:jme), INTENT(IN)     :: xlat,xlong, swdown
+   real(RKIND), DIMENSION(ims:ime,jms:jme), INTENT(IN)     :: hwp, peak_hr, fire_end_hr !RAR: Shall we make fire_end integer?
+   real(RKIND), DIMENSION(ims:ime,jms:jme), INTENT(INOUT)  :: coef_bb_dc    ! RAR:
+   real(RKIND), DIMENSION(ims:ime,jms:jme), INTENT(IN)     :: hwp_prevd
+   real(RKIND), DIMENSION(ims:ime,kms:kme,jms:jme), INTENT(IN) :: dz8w,rho_phy  !,rel_hum
+   real(RKIND), INTENT(IN) ::  dtstep, gmt
+   real(RKIND), INTENT(IN) ::  time_int, pi, ebb_min       ! RAR: time in seconds since start of simulation
+   INTEGER, DIMENSION(ims:ime,jms:jme), INTENT(IN) :: fire_type
+   integer, INTENT(IN) ::  ebb_dcycle     ! RAR: this is going to be namelist dependent, ebb_dcycle=means 
+   real(RKIND), DIMENSION(ims:ime,jms:jme), INTENT(INOUT) :: fire_hist
+!>--local 
+   logical, intent(in)  :: add_fire_moist_flux
+   integer :: i,j,k,n,m
+   integer :: icall=0
+   real(RKIND) :: conv_rho, conv, dm_smoke, dm_smoke_coarse, dc_hwp, dc_gp, dc_fn !daero_num_wfa, daero_num_ifa !, lu_sum1_5, lu_sum12_14
+   
+   INTEGER, PARAMETER :: kfire_max=51    ! max vertical level for BB plume rise
+   real(RKIND), PARAMETER :: ef_h2o=324.22  ! Emission factor for water vapor
+   ! Constants for the fire diurnal cycle calculation ! JLS - needs to be
+   ! defined below due to intent(in) of pi
+   real(RKIND) :: coef_con
+   real(RKIND) :: timeq, fire_age, age_hr, dt1,dt2,dtm         ! For BB emis. diurnal cycle calculation
+
+! For Gaussian diurnal cycle
+   real(RKIND), INTENT(IN) :: sc_factor
+   real(RKIND), PARAMETER :: rinti=2.1813936e-8, ax2=3400., const2=130., &
+                   coef2=10.6712963e-4, cx2=7200., timeq_max=3600.*24.
+!>-- Fire parameters: Fores west, Forest east, Shrubland, Savannas, Grassland, Cropland
```

Comment:
there are six types but only five numbers

Reply:
This file was updated.